packages/mojatax/src/Adaptor.js

```diff
@@ -0,0 +1,162 @@
+import { expandReferences } from '@openfn/language-common/util';
+import * as util from './Utils';
+import { execute as commonExecute } from '@openfn/language-common';
+
+/**
+ * State object
+ * @typedef {Object} MojataxHttpState
+ * @property data - The response body (as JSON)
+ * @property response - The HTTP response from the Mojatax server (excluding the body)
+ * @property references - An array of all previous data objects used in the Job
+ **/
+
+/**
+ * Options provided to Mojatax HTTP request
+ * @typedef {Object} MojataxRequestOptions
+ * @public
+ * @property {object|string} body - body data to append to the request. JSON will be converted to a string (but a content-type header will not be attached to the request).
+ * @property {object} errors - Map of errorCodes -> error messages, ie, `{ 404: 'Resource not found;' }`. Pass `false` to suppress errors for this code.
+ * @property {object} query - An object of query parameters to be encoded into the URL.
+ * @property {object} headers - An object of headers to append to the request.
+ * @property {string} parseAs - Parse the response body as json, text or stream. By default will use the response headers.
+ * @property {number} timeout - Request timeout in ms. Default: 300 seconds.
+ * @property {object} tls - TLS/SSL authentication options. See https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions
+ */
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      util.authorize,
+      ...operations
+    )({
+      ...initialState,
+      ...state,
+    });
+  };
+}
+
+/**
+ * Make a POST request to any Mojatax endpoint.
+ * @example <caption>Make a POST request to create an invoice</caption>
```

Comment:
Nice to have an example (and good caption too!) but can the payload be any smaller?

I think `invoice_id` and `customerId`, maybe with an empty items array, would suffice. Arguably I suppose any required fields should be in here. But the example is only supposed to show the pattern and maybe provide a snippet - it doesn't need to be quite so detailed :)

This comment applies to a few examples